library/core/src/iter/adapters/skip.rs

```diff
@@ -114,6 +114,17 @@ where
         }
         self.iter.fold(init, fold)
     }
+
+    #[inline]
+    fn advance_by(&mut self, n: usize) -> Result<(), usize> {
+        if self.n >= n {
+            self.n -= n;
+            return Ok(());
+        }
+        let rem = n - self.n;
+        self.n = 0;
+        self.iter.advance_by(rem)
```

Comment:
This is couting the items to skip together with the ones to advance instead of ignoring them.

Reply:
The whole method does the opposite of what it's supposed to do. 😳